_ml-commons-plugin/extensibility/bedrock-connector.md

```diff
@@ -0,0 +1,274 @@
+---
+layout: default
+title: Bedrock connector
+has_children: false
+nav_order: 63
+parent: Connectors
+grand_parent: Connecting to remote models 
+---
+
+# Bedrock connector
+
+To connect to a model on [Amazon Bedrock](https://aws.amazon.com/bedrock/), you can create a Bedrock connector. The following tutorial illustrates a [multimodal semantic search]({{site.url}}{{site.baseurl}}/search-plugins/neural-multimodal-search/) by text and image using a Bedrock model.
+
+## Prerequisites
+
+Before you start, update cluster settings with the following example configuration:
+
+```json
+PUT /_cluster/settings
+{
+  "persistent": {
+    "plugins.ml_commons.native_memory_threshold": 95,
+    "plugins.ml_commons.only_run_on_ml_node": false,
+    "plugins.ml_commons.allow_registering_model_via_url": true,
+    "plugins.ml_commons.model_access_control_enabled": true,
+    "plugins.ml_commons.trusted_connector_endpoints_regex": [
+      """^https://runtime\.sagemaker\..*[a-z0-9-]\.amazonaws\.com/.*$""",
+      """^https://api\.openai\.com/.*$""",
+      """^https://api\.cohere\.ai/.*$""",
+      """^https://bedrock\..*\.amazonaws.com/.*$"""
+    ]
+  }
+}
+```
+{% include copy-curl.html %}
+
+The example configuration ensures that you can register and deploy a remote Bedrock model in your OpenSearch cluster. For more information, see [ML Commons cluster settings]({{site.url}}{{site.baseurl}}/ml-commons-plugin/cluster-settings/) and [Using custom models within OpenSearch]({{site.url}}{{site.baseurl}}/ml-commons-plugin/ml-framework/).
+
+## Step 1: Register a model group
+
+To register a model group, send the following request:
+
+```json
+POST /_plugins/_ml/model_groups/_register
+{
+ "name": "test_model_group_public",
+ "description": "This is a public model group"
+}
+```
+{% include copy-curl.html %}
+
+The response contains the model group ID that you'll use to register a model to this model group:
+
+```json
+{
+ "model_group_id": "1jriBYsBq7EKuKzZX131",
+ "status": "CREATED"
+}
+```
+
+To learn more about model groups, see [Model access control]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control/).
+
+## Step 2: Create a Bedrock connector
+
+Create a standalone Bedrock connector by sending the following request:
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "Bedrock multimodal embedding model",
+  "description": "Bedrock multimodal model which supports both text and image embeddings",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+    "region": "us-east-1",
+    "service_name": "bedrock"
+  },
+  "credential": {
+    "access_key": "<REPLACE WITH YOUR BEDROCK ACCESS KEY>", 
+    "secret_key": "<REPLACE WITH YOUR BEDROCK SECRET KEY>"
```

Comment:
User can also choose to use temporary credential for connector https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html, then user need to add `session_token` field in `credential` part